app/jobs/yearly_review.rb

```diff
@@ -621,5 +623,34 @@ def most_bookmarked_topic_sql
         LIMIT :limit
       SQL
     end
+
+    # Title matching is for legacy purposes only, the custom field existence
+    # is the new way to check if a review topic has been created.
+    def review_topic_exists?(title, review_year)
+      DB
+        .query_single(
+          <<~SQL,
+          SELECT CASE WHEN EXISTS (
+            SELECT 1
+            FROM topics
+            WHERE title = :title AND user_id = :system_user_id AND deleted_at IS NULL
+          ) THEN 1 WHEN EXISTS (
+            SELECT 1
+            FROM topics
+            INNER JOIN topic_custom_fields ON topic_custom_fields.topic_id = topics.id
+            WHERE topic_custom_fields.name = :custom_field_name AND
+              topic_custom_fields.value = :review_year AND
+              topics.deleted_at IS NULL AND
+              user_id = :system_user_id
+          ) THEN 1 ELSE 0 END
+        SQL
+          title: title,
+          system_user_id: Discourse.system_user.id,
+          custom_field_name: ::YearlyReview::POST_CUSTOM_FIELD,
+          review_year: review_year.to_s,
+        )
+        .first
+        .positive?
```

Comment:
Oh 🤔 

I do wonder if it's worth it carrying this old "review topic title exists" check into the future, as opposed to making a migration to old topics to have the custom field, and only maintaining one check.

Reply:
Actually I don't even need this old check or a migration -- I realised that we already add `::YearlyReview::POST_CUSTOM_FIELD` to the topic. Will remove the check.